pkg/sqlite/database.go

```diff
@@ -81,19 +80,67 @@ type storeRepository struct {
 	Group          *GroupStore
 }
 
+type DatabaseType string
+
+const (
+	PostgresBackend DatabaseType = "POSTGRES"
+	SqliteBackend   DatabaseType = "SQLITE"
+)
+
+type dbInterface interface {
+	Analyze(ctx context.Context) error
+	Anonymise(outPath string) error
+	AnonymousDatabasePath(backupDirectoryPath string) string
+	AppSchemaVersion() uint
+	Backup(backupPath string) (err error)
+	Begin(ctx context.Context, writable bool) (context.Context, error)
+	Open() error
+	Close() error
+	Commit(ctx context.Context) error
+	DatabaseBackupPath(backupDirectoryPath string) string
+	DatabasePath() string
+	DatabaseType() DatabaseType
+	ExecSQL(ctx context.Context, query string, args []interface{}) (*int64, error)
+	IsLocked(err error) bool
+	Optimise(ctx context.Context) error
+	QuerySQL(ctx context.Context, query string, args []interface{}) ([]string, [][]interface{}, error)
+	ReInitialise() error
+	Ready() error
+	Remove() error
+	Repository() models.Repository
+	Reset() error
+	RestoreFromBackup(backupPath string) error
+	Rollback(ctx context.Context) error
+	RunAllMigrations() error
+	SetBlobStoreOptions(options BlobStoreOptions)
+	Vacuum(ctx context.Context) error
+	Version() uint
+	WithDatabase(ctx context.Context) (context.Context, error)
+	getDatabaseSchemaVersion() (uint, error)
+	initialise() error
+	lock()
+	needsMigration() bool
+	open(disableForeignKeys bool, writable bool) (conn *sqlx.DB, err error)
+	openReadDB() error
+	openWriteDB() error
+	txnComplete(ctx context.Context)
+	unlock()
+}
+
 type Database struct {
 	*storeRepository
+	dbInterface
 
-	readDB  *sqlx.DB
-	writeDB *sqlx.DB
-	dbPath  string
+	readDB   *sqlx.DB
+	writeDB  *sqlx.DB
+	dbConfig interface{}
```

Comment:
Looks like this is a `string` in both implementations, so could be reverted to type string?

Reply:
The reason this was done was to hopefully allow for other db implementations, as i dont know what specific permissions they might want, so i changed it to interface.
Im not against changing it to string.

Reply:
True, but in any case I think this could be a string. The underlying DB driver will decide if they want to parse it or not. It's hard to pass structured properties as config vars, and uncommon.